Add environment configuration guide

gioisco opened the pull request:
**Summary:**

I've created a comprehensive guide with 10 steps to help developers set up their environment for building and running the sossoldi app. 
Currently tailored for Linux OS, particularly Pop!_OS, and utilizes the IntelliJ Community Edition as the preferred IDE.

Feel free to enhance the guide with additional configurations to cover diverse environments.

**Changes:**
Added DEVELOPER_GUIDE.md
Included a screenshot, for better understand how to install `cmdline-tools` (Screenshot_Intellij_cmdline-tools.png)

You can see the [guide here](https://github.com/gioisco/sossoldi/blob/developer-guide/DEVELOPER_GUIDE.md)

theperu commented on DEVELOPER_GUIDE.md:
```diff
@@ -0,0 +1,305 @@
+# Dev Environment Configuration
```
Since this is specific for Linux I would change the title to "Linux Dev Environment Configuration" and I would do the same thing for the file name

theperu commented on Screenshot_Intellij_cmdline-tools.png:
I would move this in the asset folder

gioisco replied:
I appreciate the feedback. I will proceed to make the necessary changes accordingly.

theperu reviewed (changes requested):
Thank you! This documentation will definitely be helpful in the future, I made some suggestion here and there but it seems good for me.
Since we are moving to GitHub Pages for the documentation I think this PR will remain open until that is merged, then we will move this in the _docs_ folder. If you never used GitHub Pages don't worry they fully support .md files so you don't need to make specific changes for it 😁

gioisco commented:
Thank you for your feedback! I'm glad to hear that the documentation is useful.

Regarding the move to GitHub Pages, I'm completely on board. I haven't used GitHub Pages before, but I'm aware that they fully support .md files, so I intentionally wrote the guide with that in mind. Looking forward to any further guidance during the transition!

😊🚀

theperu commented:
Hi! We enabled the page so now you can go forward with the changes and then we will close and merge this PR. As I told you you'll have to move all the changes that you made to the _/docs_ folder, all the images should go in the _/docs/assets_ folder. You will find more info on how to check the page in your repo and other parts that we want to add in the issue #123

theperu commented:
Hi @gioisco did you have time to look at the changes that I suggested? Let me know if you need anything 😁

gioisco commented:
I've implemented the requested changes. 

Additionally, I fixed some misspellings, improved the titles of sections, and moved the files under the 'docs' directory (the screenshot is now in the 'assets' directory). 

Finally, I included a link to the Linux setup guide in the 'Useful Links' section on the GitHub Page home (the README under the 'docs' directory).

theperu reviewed (approved):
LGTM 👍🏻
